src/bin/apigw.rs

```diff
@@ -0,0 +1,84 @@
+use lambda_http::{http::StatusCode, IntoResponse, Request, RequestExt, Response, Error, service_fn};
+use serde_derive::Deserialize;
+use serde_json::json;
+use tracing::{event, Level};
+use url::Url;
+
+use s3_rust_noodles_bam::{read_bam_header, stream_s3_object_with_params};
+use s3_rust_noodles_bam::telemetry::{get_subscriber, init_subscriber};
+
+#[derive(Debug, Deserialize, Default)]
+struct EventPayload {
+    #[serde(default)]
+    bam: String,
+    region: Option<String>,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let subscriber = get_subscriber("get_bam_header".into(), "info".into(), std::io::stdout);
+    init_subscriber(subscriber);
+
+    lambda_http::run(service_fn(get_bam_header)).await?;
+    Ok(())
+}
+
+async fn get_bam_header(request: Request) -> Result<impl IntoResponse, Error> {
```

Comment:
I've got this setup as returning a plain `Response<Body>` in htsget-http-lambda, with any errors wrapped up in the response itself, rather than a `Result`.

Does `lambda_http::run` convert errors into responses?

Reply:
> Does lambda_http::run convert errors into responses?

I think, it does it so. Check `cargo lambda new --help` init template code.

Reply:
There has been quite a big torrent of commits on the lambda runtime repo in the last 2-3 weeks, worth checking out the changes w.r.t Body type and other improvements.

Reply:
> I've got this setup as returning a plain Response<Body> in htsget-http-lambda, with any errors wrapped up in the response itself, rather than a Result.

Interesting, @mmalenic, I'm not sure if `Response<Body>` alone can do "proper" error management when unwinding. Most of the official examples (https://github.com/awslabs/aws-lambda-rust-runtime/tree/main/lambda-http/examples) return the type Victor pointed out or `Result<Response<Body>, Error>` instead.

Reply:
Btw, on-topic ongoing draft PR: https://github.com/awslabs/aws-lambda-rust-runtime/pull/466

Reply:
All the examples appear to use a function that returns `Ok`, even when the response uses an 'error' status code such as [here](https://github.com/awslabs/aws-lambda-rust-runtime/blob/aff310b0e52d8f4a389b98b672a7121b230fbe82/lambda-http/examples/shared-resources-example.rs#L28).

The actual error seems to get returned to the lambda api [here](https://github.com/awslabs/aws-lambda-rust-runtime/blob/aff310b0e52d8f4a389b98b672a7121b230fbe82/lambda-runtime/src/lib.rs#L146). And then I guess its up to api gateway on how to respond?